src/main/java/duke/command/DoneCommand.java

```diff
@@ -0,0 +1,59 @@
+package duke.command;
+
+import duke.exception.BadInputFormatException;
+import duke.exception.NoSuchTaskException;
+import duke.util.Storage;
+import duke.util.TaskList;
+
+/** Represents the "done" command. */
+public class DoneCommand extends Command {
+    /** The index of the Task to be marked as done. */
+    private int index;
+
+    /**
+     * DoneCommand constructor.
+     *
+     * @param index The index of the Task to be marked as done.
+     */
+    private DoneCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * DoneCommand factory method.
+     *
+     * @param index The index of the Task to be deleted as a numeric string.
+     * @return A DoneCommand object.
+     * @throws BadInputFormatException If the provided input string is not a numeric.
+     */
+    public static DoneCommand of(String index) throws BadInputFormatException {
+        try {
+            return new DoneCommand(Integer.parseInt(index) - 1);
+        } catch (NumberFormatException e) {
+            throw new BadInputFormatException();
+        }
+    }
+
+    /**
+     * Marks the Task at the specified index as done.
+     *
+     * @param tasks The list of tasks in the program.
+     * @param storage The storage utility.
+     * @throws NoSuchTaskException If the task does not exist at the specified index.
```

Comment:
Maybe you could consider adding a description of what the method returns under Javadocs? I think you may have missed it out! You could consider making this change to all your command classes (classes that inherit from Command class) :)

Reply:
Ah, I accidentally left this out after implementing my GUI. Thanks for pointing it out :)